frontend/src/pages/PipelineDetails.tsx

```diff
@@ -235,24 +239,49 @@ class PipelineDetails extends Page<{}, PipelineDetailsState> {
     if (origin) {
       const msgRunOrRecurringRun = origin.isRecurring ? 'recurring run' : 'run';
       try {
+        // TODO(jlyaoyuli): change to v2 API after v1 is deprecated
         if (origin.isRecurring) {
-          origin.recurringRun = await Apis.jobServiceApi.getJob(origin.recurringRunId!);
+          origin.v1RecurringRun = await Apis.jobServiceApi.getJob(origin.recurringRunId!);
+          origin.v2RecurringRun = await Apis.recurringRunServiceApi.getRecurringRun(
+            origin.recurringRunId!,
+          );
         } else {
-          origin.run = await Apis.runServiceApi.getRun(origin.runId!);
+          origin.v1Run = await Apis.runServiceApi.getRun(origin.runId!);
+          origin.v2Run = await Apis.runServiceApiV2.getRun(origin.runId!);
+        }
+
+        // v2 run or recurring run with pipeline_version_id
+        const pipelineVersionIdFromOrigin = origin.isRecurring
+          ? origin.v2RecurringRun?.pipeline_version_id
+          : origin.v2Run?.pipeline_version_id;
```

Comment:
We should remember the decision here so you don't need to check for both `origin.v2Run?.pipeline_spec` and `origin.v2RecurringRun?.pipeline_spec` in the logic below.